internal/response-consumer/handler.go

```diff
@@ -164,6 +156,100 @@ func (this *handler) onMessage(ctx context.Context, msg *k.Message) {
 	}
 }
 
+func buildCaseExpr(caseOptions []string) clause.Expr {
+	argsLen := len(caseOptions)
+	caseExprStr := "CASE "
+	varsArray := make([]interface{}, argsLen)
+
+	for i, option := range caseOptions {
+		varsArray[i] = clause.Expr{SQL: option}
+
+		if i == argsLen-1 {
+			caseExprStr += fmt.Sprintf("ELSE (?) END")
+		} else if i%2 == 0 {
+			caseExprStr += fmt.Sprintf("WHEN (?) ")
+		} else {
+			caseExprStr += fmt.Sprintf("THEN (?) ")
+		}
+	}
+
+	return clause.Expr{
+		SQL:  caseExprStr,
+		Vars: varsArray,
+	}
+
+}
+
+func satAssignmentWithCase(responseFull bool, updateHost db.RunHost) map[string]interface{} {
+	satSequence, status, log := *updateHost.SatSequence, updateHost.Status, updateHost.Log
+
+	satNullClause := "sat_sequence IS NULL"
+
+	statusCase := []string{
+		fmt.Sprintf("%s OR sat_sequence < %d", satNullClause, satSequence),
+		fmt.Sprintf("'%s'", status),
+		"status",
+	}
+
+	satSeqCase := []string{
+		fmt.Sprintf("%s OR sat_sequence < %d", satNullClause, satSequence),
+		strconv.Itoa(satSequence),
+		"sat_sequence",
+	}
+
+	logCase := []string{
+		fmt.Sprintf("%s OR sat_sequence < %d", satNullClause, satSequence),
+		fmt.Sprintf("'%s'", log),
+		"log",
+	}
+
+	if !responseFull {
+		logCase = []string{
+			fmt.Sprintf("(%s AND %d > 0) OR sat_sequence + 1 < %d", satNullClause, satSequence, satSequence),
+			fmt.Sprintf("log || '&#8230;' || '%s'", log),
+			fmt.Sprintf("sat_sequence > %d", satSequence),
+			"log",
+			fmt.Sprintf("log || '%s'", log),
+		}
+	}
+
+	return map[string]interface{}{
+		"status":       buildCaseExpr(statusCase),
+		"sat_sequence": buildCaseExpr(satSeqCase),
+		"log":          buildCaseExpr(logCase),
+	}
+}
+
+func satUpdateRecord(ctx context.Context, tx *gorm.DB, responseFull bool, toUpdate []db.RunHost) error {
+	for _, runHost := range toUpdate {
+		updateResult := tx.Model(db.RunHost{}).
+			Where("run_id = ? AND host = ?", runHost.RunID, runHost.Host).
```

Comment:
satellite hosts should be matched based on the inventory_id field, not the host identifier.